mne/morph.py

```diff
@@ -0,0 +1,1443 @@
+                                        subjects_dir, 1)
+            data_from = _check_hemi_data(data_from, data_to)
+
+        if data_to is None:
+            raise ValueError('Please specify target to morph to.')
+
+        # check emptiness of hemispheres
+        data_to = _check_hemi_data(data_to, data_from)
+
+        # pre-compute morph matrix
+        morph_mat = _compute_morph_matrix(
+            subject_from=morph.subject_from,
+            subject_to=morph.subject_to,
+            vertices_from=data_from,
+            vertices_to=data_to,
+            subjects_dir=subjects_dir,
+            smooth=morph.smooth,
+            warn=morph.warn,
+            xhemi=morph.xhemi,
+            verbose=verbose)
+        data.update({'morph_mat': morph_mat, 'vertno': data_to})
+
+    return data
+
+
+def _interpolate_data(stc, morph_data, mri_resolution=True, mri_space=True,
+                      format='nifti1'):
+    """Interpolate source estimate data to MRI."""
+    if format != 'nifti1' and format != 'nifti2':
+        raise ValueError("invalid format specifier %s. Must be 'nifti1' or"
+                         " 'nifti2'" % format)
+    if format == 'nifti1':
+        from nibabel import (Nifti1Image as NiftiImage,
+                             Nifti1Header as NiftiHeader)
+    elif format == 'nifti2':
+        from nibabel import (Nifti2Image as NiftiImage,
+                             Nifti2Header as NiftiHeader)
+    from dipy.align.reslice import reslice
+
+    # setup volume parameters
+    n_times = stc.data.shape[1]
+    shape3d = morph_data['src_shape']
+    shape = (n_times,) + shape3d
+    vol = np.zeros(shape)
+
+    voxel_size_defined = False
+
+    if isinstance(mri_resolution, (int, float)) and not isinstance(
+            mri_resolution, bool):
+        # use iso voxel size
+        mri_resolution = (float(mri_resolution),) * 3
+
+    if isinstance(mri_resolution, tuple):
+        voxel_size = mri_resolution
+        voxel_size_defined = True
+        mri_resolution = True
+
+    # if data wasn't morphed yet - necessary for call of
+    # stc_unmorphed.as_volume. Since only the shape of src is known, it cannot
+    # be resliced to a given voxel size without knowing the original.
+    if 'morph_shape' not in morph_data and voxel_size_defined:
+        raise ValueError(
+            "Cannot infer original voxel size for reslicing... "
+            "set mri_resolution to boolean value or apply morph first.")
+
+    # use mri resolution as represented in src
+    if mri_resolution:
+        mri_shape3d = morph_data['src_shape_full']
+        mri_shape = (n_times,) + mri_shape3d
+        mri_vol = np.zeros(mri_shape)
+        interpolator = morph_data['interpolator']
+
+    mask3d = morph_data['inuse'].reshape(shape3d).astype(np.bool)
+    n_vertices = np.sum(mask3d)
+
+    n_vertices_seen = 0
+    for k, v in enumerate(vol):  # loop over time instants
```

Comment:
Single letter variable names are considered bad for readability. You can't search for them using Ctrl + F. And also, they are not descriptive